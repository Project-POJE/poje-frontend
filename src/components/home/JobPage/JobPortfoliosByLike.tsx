import { JobPortfoliosContainer } from './common/commomJobStyledComponents';

export default function JobPortfoliosByLike() {
  return <JobPortfoliosContainer></JobPortfoliosContainer>;
}
