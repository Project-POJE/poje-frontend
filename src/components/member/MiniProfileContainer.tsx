export default function MiniProfileContainer() {
  return <div></div>;
}
