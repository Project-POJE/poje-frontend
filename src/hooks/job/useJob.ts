export default function useJob() {
  return 123;
}
