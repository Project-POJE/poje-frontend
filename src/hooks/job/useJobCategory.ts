import service from '@src/service';
import { useQuery } from '@tanstack/react-query';

export default function useJobCategory() {
  const {
    data: category = [],
    isLoading,
    isSuccess,
  } = useQuery(
    ['category'],
    () => {
      return service.job.getJobCategory();
    },
    {
      onError: (e: unknown) => {
        console.log('에러에요에어에어에어에어에어에어에어');
      },
    }
  );
  return { category, isLoading, isSuccess };
}
