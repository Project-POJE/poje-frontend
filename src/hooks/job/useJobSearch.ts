import { useCallback } from 'react';
export default function useJobSearch() {
  const prefetchJobSearch = useCallback((keyword: string) => {
    //Todo 검색결과 프리패치 함수
    console.log('프리패칭 수행중 ...');
  }, []);

  const getJobListByKeyword = useCallback((keyword: string) => {
    //Todo 검색결과 반환하기
  }, []);
  return { prefetchJobSearch, getJobListByKeyword };
}
