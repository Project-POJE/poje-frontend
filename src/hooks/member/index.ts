import service from '@src/service';
import { useQuery } from '@tanstack/react-query';

export const usePortfolioListsByLike = (page: string) => {
  const { data } = useQuery(
    ['portfoliosListWithLike', page],
    () => service.job.getePortfolioCardsWithLike({ page }),
    { staleTime: 1000 * 60, suspense: true }
  );

  return { pageingUtil: data!.paging, pfAndMemberResp: data!.pfAndMemberResp };
};
