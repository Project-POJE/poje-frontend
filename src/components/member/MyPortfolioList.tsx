import { useGetMyPortfolios } from '@src/hooks/member';
import styled from 'styled-components';
import ProfileItemRemaster from '../home/JobPage/ProfileItemRemaster';
import usePortfolioCRUD from '@src/hooks/portfolio/usePortfolioCRUD';

export default function MyPortfolioList() {
  const list = useGetMyPortfolios();
  const { onDeletePortfolio } = usePortfolioCRUD();

  return (
    <>
      {list.map((item) => (
        <Test key={item.portfolioId + item.nickName}>
          <ProfileItemRemaster
            {...item}
            onDeletePortfolio={onDeletePortfolio}
          />
        </Test>
      ))}
    </>
  );
}

const Test = styled.div`
  flex-shrink: 0;
  flex-basis: 240px;
`;
