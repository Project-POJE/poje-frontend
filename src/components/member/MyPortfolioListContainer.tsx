import { Suspense } from 'react';
import styled from 'styled-components';
import LoadingSpiner from '../portfolio/common/LoadingSpiner';
import MyPortfolioList from './MyPortfolioList';

export default function MyPortfolioListContainer() {
  return (
    <Container>
      <Category>내 포트폴리오 리스트</Category>
      <FlexList>
        <Suspense fallback={<LoadingSpiner text={'로딩중'} />}>
          <MyPortfolioList />
        </Suspense>
      </FlexList>
    </Container>
  );
}

const Container = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const Category = styled.h2`
  color: ${({ theme }) => theme.textColor};
`;

const FlexList = styled.div`
  display: flex;
  overflow-x: auto;
  flex-wrap: nowrap;
  gap: 0.8rem;
  padding: 2rem;
`;
