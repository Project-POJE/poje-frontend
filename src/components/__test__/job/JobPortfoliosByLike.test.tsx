import JobPortfoliosByLike from '@src/components/home/JobPage/JobPortfoliosByLike';
import { usePortfolioListsByLike } from '@src/hooks/member';
import { mockAPIPortfolioList } from '@src/test_util/mock/data/job';
import { render } from '@testing-library/react';
import { useParams } from 'react-router-dom';

jest.mock('@src/hooks/member');
jest.mock('react-router-dom');

const mockUsePortfolioListsByLike: jest.Mock =
  usePortfolioListsByLike as jest.MockedFunction<any>;
const mockUseParams: jest.Mock = useParams as jest.MockedFunction<any>;

describe('JobPortfoliosByLike', () => {
  it('스냅샷 테스트', () => {
    mockUsePortfolioListsByLike.mockImplementation(() => ({
      pageingUtil: mockAPIPortfolioList.paging,
      pfAndMemberResp: mockAPIPortfolioList.pfAndMemberResp,
    }));
    mockUseParams.mockImplementation(() => ({ page: 1 }));

    const { asFragment } = render(<JobPortfoliosByLike />);

    expect(asFragment()).toMatchSnapshot();
  });
});
