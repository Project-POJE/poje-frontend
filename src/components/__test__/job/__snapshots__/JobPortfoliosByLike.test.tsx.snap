// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobPortfoliosByLike 스냅샷 테스트 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe fzPIWk"
  >
    <div
      class="sc-gswNZR dEeHfy"
    >
      <ul
        class="sc-csuSiG lcITfR"
      >
        <li
          class="sc-hLBbgP hfyZwr"
        >
          <div
            class="sc-eDvSVe dFcDTk"
          >
            <img
              alt="명수의 포트폴리오"
              class="sc-jSUZER cLkbEL"
              src="https://pojeawsbucket.s3.ap-northeast-2.amazonaws.com/portfolio/d75081b6-d43a-4fbf-aaa3-03dcc80ca4d1.PNG"
            />
          </div>
          <div
            class="sc-gKPRtg enTVrJ"
          >
            <h2
              class="sc-iBYQkv jVHjpa"
            >
              명수의 포트폴리오
            </h2>
            <p
              class="sc-ftTHYK cvPSBb"
            >
              예비 백엔드 개발자의 포트폴리오입니다.
            </p>
          </div>
          <footer
            class="sc-pyfCe ceXemT"
          >
            <div
              class="sc-jrcTuL leOpnW"
            >
              <img
                class="sc-ipEyDJ hYbAXg"
                src="https://pojeawsbucket.s3.ap-northeast-2.amazonaws.com/profile/9ab46f45-3431-4098-a4da-69c2c21f877d.PNG"
              />
              <p>
                싸르
              </p>
            </div>
            <div
              class="sc-kDvujY hlFFDt"
            >
              <svg
                class="like"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9zM512 814.8S156 586.7 156 385.5C156 283.6 240.3 201 344.3 201c73.1 0 136.5 40.8 167.7 100.4C543.2 241.8 606.6 201 679.7 201c104 0 188.3 82.6 188.3 184.5 0 201.2-356 429.3-356 429.3z"
                />
                <path
                  d="M679.7 201c-73.1 0-136.5 40.8-167.7 100.4C480.8 241.8 417.4 201 344.3 201c-104 0-188.3 82.6-188.3 184.5 0 201.2 356 429.3 356 429.3s356-228.1 356-429.3C868 283.6 783.7 201 679.7 201z"
                />
              </svg>
              <p>
                2
              </p>
            </div>
          </footer>
        </li>
        <li
          class="sc-hLBbgP hfyZwr"
        >
          <div
            class="sc-eDvSVe dFcDTk"
          >
            <img
              alt="승윤의 포트폴리오"
              class="sc-jSUZER cLkbEL"
              src="https://pojeawsbucket.s3.ap-northeast-2.amazonaws.com/portfolio/d75081b6-d43a-4fbf-aaa3-03dcc80ca4d1.PNG"
            />
          </div>
          <div
            class="sc-gKPRtg enTVrJ"
          >
            <h2
              class="sc-iBYQkv jVHjpa"
            >
              승윤의 포트폴리오
            </h2>
            <p
              class="sc-ftTHYK cvPSBb"
            >
              예비 백엔드 개발자의 포트폴리오입니다.
            </p>
          </div>
          <footer
            class="sc-pyfCe ceXemT"
          >
            <div
              class="sc-jrcTuL leOpnW"
            >
              <img
                class="sc-ipEyDJ hYbAXg"
                src="https://pojeawsbucket.s3.ap-northeast-2.amazonaws.com/profile/9ab46f45-3431-4098-a4da-69c2c21f877d.PNG"
              />
              <p>
                싸르
              </p>
            </div>
            <div
              class="sc-kDvujY hlFFDt"
            >
              <svg
                class="like"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9zM512 814.8S156 586.7 156 385.5C156 283.6 240.3 201 344.3 201c73.1 0 136.5 40.8 167.7 100.4C543.2 241.8 606.6 201 679.7 201c104 0 188.3 82.6 188.3 184.5 0 201.2-356 429.3-356 429.3z"
                />
                <path
                  d="M679.7 201c-73.1 0-136.5 40.8-167.7 100.4C480.8 241.8 417.4 201 344.3 201c-104 0-188.3 82.6-188.3 184.5 0 201.2 356 429.3 356 429.3s356-228.1 356-429.3C868 283.6 783.7 201 679.7 201z"
                />
              </svg>
              <p>
                2
              </p>
            </div>
          </footer>
        </li>
      </ul>
    </div>
    <ul
      class="sc-eDWCr gA-DMfG"
    >
      <li
        class="sc-bqWxrE fYSHEI"
        data-testid="prev-end-btn"
      >
        &lt;&lt;
      </li>
      <li
        class="sc-ksBlkl jLcwFf"
      >
        1
      </li>
      <li
        class="sc-bqWxrE fYSHEI"
        data-testid="next-end-btn"
      >
        &gt;&gt;
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
